source/Dgraph/Transactions/ITransaction.cs

```diff
@@ -14,74 +14,83 @@
  * limitations under the License.
  */
 
-using System;
-using System.Threading.Tasks;
+using FluentResults;
 using Grpc.Core;
 
 namespace Dgraph.Transactions
 {
-
     /// <summary>
-    /// Use transactions like :
-    ///     
-    /// <code>  
-    /// using(var txn = client.NewTransaction()) {
-    ///    txn.mutate
-    ///    txn.query
-    ///    txn.mutate
-    ///    txn.commit
-    /// } 
-    /// </code>
-    ///
-    /// or
-    ///
-    /// <code>  
-    /// var txn = client.NewTransaction()
-    /// txn.mutate()
-    /// if(...) {
-    ///    txn.commit();
-    /// } else {
-    ///    txn.discard();
-    /// }
-    /// </code>
+    /// A single atomic transaction.
+    /// A transaction lifecycle is as follows:
+    ///  1. Created using IDgraphClient.NewTransaction()
+    ///  2. Various Query and Mutate calls made.
+    ///  3. Commit or Discard used. If any mutations have been made, It's important
+    ///     that at least one of these methods is called to clean up resources. Discard
+    ///     is a no-op if Commit has already been called.
     /// </summary>
     public interface ITransaction : IQuery, IDisposable
     {
+        /// <summary>
+        /// Mutate allows data stored on Dgraph instances to be modified.
+        /// The fields in api.Mutation come in pairs, set and delete.
+        /// Mutations can either be encoded as JSON or as RDFs.
+        ///
+        /// If CommitNow is set, then this call will result in the transaction
+        /// being committed. In this case, an explicit call to Commit doesn't
+        /// need to be made subsequently.
+        ///
+        /// If the mutation fails, then the transaction is discarded and all
+        /// future operations on it will fail.
+        /// </summary>
+        Task<Result<Response>> Mutate(Api.Mutation mutation, CallOptions? options = null)
+        {
+            var request = new Api.Request();
+            request.Mutations.Add(mutation);
+            request.CommitNow = mutation.CommitNow;
+            return Do(request, options);
+        }
+
+        /// <summary>
+        /// Mutate allows data stored on Dgraph instances to be modified.
```

Comment:
we should just say here that it is same as the other Mutate, just that arg type is different.